packages/middleware/src/index.ts

```diff
@@ -52,24 +53,44 @@ async function getMiddlewareFile(workingDirectory: string) {
 }
 
 export async function build({ workPath }: { workPath: string }) {
-  const entriesPath = join(workPath, ENTRIES_NAME);
+  const entriesPath = join(workPath, TMP_ENTRIES_NAME);
+  const transientFilePath = join(workPath, TMP_MIDDLEWARE_BUNDLE);
   const middlewareFile = await getMiddlewareFile(workPath);
   if (!middlewareFile) return;
 
   console.log('Compiling middleware file: %j', middlewareFile);
 
-  // Create `_ENTRIES` wrapper
-  await fsp.copyFile(join(__dirname, 'entries.js'), entriesPath);
-
-  // Build
+  /**
+   * Two builds happen here, because esbuild doesn't offer a way to add a banner
+   * to individual input files, and the entries wrapper relies on running in
+   * non-strict mode to access the ENTRIES global.
+   *
+   * To work around this, we bundle the middleware directly and add
+   * 'use strict'; to make the entire bundle run in strict mode. We then bundle
+   * a second time, adding the global ENTRIES wrapper and preserving the
+   * 'use strict' for the entire scope of the original bundle.
```

Comment:
Open to alternative approaches here.  I hit a snag when trying to bundle w/ swc and @leo recommended sticking w/ esbuild for now if it means we can quickly address the linked issue.

Reply:
Exactly, we definitely should switch to SWC long-term, but to make it work, we can go this route already.

Reply:
To be clear, looking for feedback (if any) on the approach of double bundling here as described in the comment.

Reply:
Maybe @TooTallNate or @styfle have a suggestion?

Reply:
Can we avoid accessing the "ENTRIES global"?

Reply:
@styfle that's a good question... my understanding was that the runtime expects it to be available to hook up the middleware. Is there another option?

Reply:
We could swap `esbuild` for `@vercel/nft` to avoid bundling files. This is how we do it for node lambdas: https://github.com/vercel/vercel/blob/adb284519ad2b424651d420cd2f46b511173044f/packages/node/src/index.ts#L193

Reply:
I'm going to look into `@vercel/nft` approach to ensure it works in prod

Reply:
Slack conversations indicated that this wouldn't be possible as production runtime doesn't support import/export. I believe that bundle to a single file is mandatory.